#!/usr/bin/python3

class CveException(BaseException):
    """
    The parent exception class of the cve tracking tool
    """

    def __init__(self, msg):
        super(CveException, self).__init__()
        self.message = msg

    def __repr__(self):
        return self.message


class ConfigNotFoundError(CveException):
    """
    Configuration file not found exception
    """

    def __init__(self, msg):
        super(ConfigNotFoundError, self).__init__(f'Configuration file not found: {msg}')


class InputError(CveException):
    """
    Input params error
    """

    def __init__(self, msg):
        super(InputError, self).__init__(f'Input parameter error: {msg}')


class RequestError(CveException):
    """
    Error calling URL
    """

    def __init__(self, msg):
        super(RequestError, self).__init__(f'Error calling URL: {msg}')


class PathNotExistError(CveException):
    """
    Path not exist error
    """

    def __init__(self, msg):
        super(PathNotExistError, self).__init__(f'Path not exist: {msg}')


class PathEmptyError(CveException):
    """
    Path is empty error
    """

    def __init__(self, msg):
        super(PathEmptyError, self).__init__(f'Path is empty: {msg}')
    
class RateLimitExceededError(CveException):

    def __init__(self):
        super(RateLimitExceededError, self).__init__(f'Rate limit exceeded. Retrying...')
